tests/jvm/src/test/scala/munit/TestSelectorSuite.scala

```diff
@@ -0,0 +1,122 @@
+package munit
+
+import sbt.testing.Event
+import sbt.testing.EventHandler
+import sbt.testing.Selector
+import sbt.testing.Status
+import sbt.testing.SuiteSelector
+import sbt.testing.TaskDef
+import sbt.testing.TestSelector
+import scala.collection.mutable
+import sbt.testing.Runner
+
+/**
+ * Dummy test suite which is needed by TestSelectorSuite
+ */
+class MyTestSuite extends FunSuite {
+
+  test("testFoo") {
+    assert(true, "Test should pass")
+  }
+
+  test("testBar") {
+    assert(true, "Test should pass")
+  }
+}
+
+/**
+ * Check if TestSelector's are correctly handled by Munit.
+ * Execute prepared TaskDef's using manually created instances of sbt.testing.{Framework and Runner}.
+ */
+class TestSelectorSuite extends FunSuite {
+  val framework = new Framework();
+  val runner: Runner = framework.runner(
+    Array.empty,
+    Array.empty,
+    this.getClass().getClassLoader()
+  );
+
+  val fingerprint = framework.munitFingerprint
+
+  /**
+   * Reference to the collection which will contain fully qualified names of executed tests.
+   * Only after finished execution it's safe to convert this mutable collection to immutable one.
+   */
+  private def getEventHandler(): (mutable.ListBuffer[String], EventHandler) = {
+    val executedItems = new mutable.ListBuffer[String]
+    val eventHandler = new EventHandler {
+      override def handle(event: Event) =
+        if (event.status() == Status.Success) {
+          executedItems += event.fullyQualifiedName()
+        }
+    }
+    (executedItems, eventHandler)
+  }
```

Comment:
Each executed test will send an event, so I use that fact to gather information about executed tests.